### PR TITLE
Test doubles

### PR DESCRIPTION
**Pull request by Patryk-Kuchta**
Your goal is to write tests for app.js using app_test.js as a starting point.

app.js interprets game commands such as "add player Alice" or "print", and uses them to update the game and save or load files.

It has three dependencies:

    the league

    the league renderer

    the file service.

app_test.js already has a test for print. Add to this by writing tests for the following commands:

    add player Alice

    record win Alice Bob

    winner

    save some/file/path

    load some/file/path

Decide whether to use stubs, mocks or the real thing for each of the dependencies. Think about what edge cases to test as well as valid commands.

## Summary by CodeRabbit

- **Tests**
	- Enhanced test coverage for command processing functionality in the application.
	- Added test cases for league state printing, player addition, win recording, winner display, and league state saving/loading.
	- Introduced a new export for the `createAndPopulateLeague` function to improve usability in the test suite.

### CODE REVIEW

**Comment by Patryk-Kuchta** [comment]
@coderabbitai review

File: test/app_test.js [hunk 1]
```diff
@@ -15,4 +18,71 @@ describe('app command processing', function () {
     const game = app.startGame(league);
     expect(game.sendCommand('print')).to.equal('rendered league');
   });
+
+  it('allows for adding a new player', function () {
+    const league = gameState.createLeague();
```
Comment (Patryk-Kuchta) [inline test/app_test.js:23]: [AI GENERATED, POSTED THROUGH MCP] Great test for adding a new player! Consider adding a test case for adding a player with a name containing spaces.

File: test/app_test.js [hunk 2]
```diff
@@ -15,4 +18,71 @@ describe('app command processing', function () {
...
+    const game = app.startGame(league);
+
+    game.sendCommand(`add player ${newPlayer}`);
+
+    mockedLeague.verify();
+  });
+
+  it('allows for recording the win', function () {
+    const league = gameState.createLeague();
+    const winner = 'Alice';
```
Comment (Patryk-Kuchta) [inline test/app_test.js:37]: [AI GENERATED, POSTED THROUGH MCP] Typo: 'looser' should be 'loser' in the variable name.

File: test/app_test.js [hunk 3]
```diff
@@ -15,4 +18,71 @@ describe('app command processing', function () {
...
+    const game = app.startGame(league);
+
+    expect(game.sendCommand('winner')).to.equal(winner);
+  });
+
+  it('allows for saving the state to specified path', function () {
+    const players = [['Alice'], ['Bob']];
+    const league = createAndPopulateLeague(players);
+
+    const path = 'some/file/path';
```
Comment (Patryk-Kuchta) [inline test/app_test.js:64]: [AI GENERATED, POSTED THROUGH MCP] Good use of mocks. Consider adding a test case for error handling when the file doesn't exist.

File: test/league_test.js [hunk 4]
```diff
@@ -13,6 +13,8 @@ function createAndPopulateLeague (playerList) {
   return league;
```
Comment (Patryk-Kuchta) [inline test/league_test.js:13]: [AI GENERATED, POSTED THROUGH MCP] Consider adding JSDoc comments to document the helper function.

**Review by Patryk-Kuchta — COMMENTED** [review]
[AI GENERATED, POSTED THROUGH MCP] Thank you for the PR! I've reviewed the test cases and have some feedback to share.
